src/db/visitor.test.ts

```diff
@@ -0,0 +1,66 @@
+import {expect} from '@esm-bundle/chai';
+import * as dag from '../dag/mod';
+import {
+  addGenesis,
+  addIndexChange,
+  addLocal,
+  addSnapshot,
+  Chain,
+} from './test-helpers';
+import {initHasher} from '../hash';
+import type {DataNode} from '../btree/node';
+import type {ReadonlyJSONValue} from '../json';
+import {Visitor} from './visitor';
+import type {Commit} from './commit';
+
+setup(async () => {
+  await initHasher();
+});
+
+test('test that we get to the data nodes', async () => {
```

Comment:
It's unclearer to me if this test exercises:
1. logic around weak hash for snapshot commits
2. logic around weak originalHash for local commits

Reply:
I added a test for weak originalHash

the weakness of snapshot commits means that things gets collected on commit so the chunk it references does not exist any more.